### ISSUE TITLE
Ability to enable Z-hop only when travelling over ironed surfaces

### ISSUE THREAD

**Issue by melvinmicallef** [supermerill/SuperSlicer#2118]
**Is your feature request related to a problem? Please describe.**
Yes, I use the top ironing pattern or ironing post processing a lot. A problem is that when there is a model which has multiple top layers, they are ironed but afterwards, the nozzle scratches over the ironed surfaces and thus, ruining them. A workaround I'm using is to enable z hop for all the print but this adds print time and sometimes reduces accuracy because of Z backlash.

**Describe the solution you'd like**
Adding an option to this dropdown that is labelled something like 'Only when crossing ironed(both infill and post processing) surfaces'.
![image](https://user-images.githubusercontent.com/60214993/147353631-7d008084-38a1-411e-8b38-716508d87047.png)

**Describe how it would work**
The slices will be able to detect if the print head is travelling over a top surface that has ironing finished and introduce zhop to that travel move

**Describe alternatives you've considered**
An alternative i see is for the nozzle to take a different route rather than z hopping, however, this might not always be possible since an unfinished part might be surrounded by  an ironed surface and would make re routing impossible. I see z hopping as the best solution

**Additional context**
An example is this model, where the red is an ironed top layer, and the blue arrow is the scratch that will occur on that surface due to zhop being off 
![image](https://user-images.githubusercontent.com/60214993/147353542-dd42cc68-2834-4d20-8377-768c4aaf9068.png)

**Comment by DVSVIDEO** [comment]
> **Is your feature request related to a problem? Please describe.** Yes, I use the top ironing pattern or ironing post processing a lot. A problem is that when there is a model which has multiple top layers, they are ironed but afterwards, the nozzle scratches over the ironed surfaces and thus, ruining them. A workaround I'm using is to enable z hop for all the print but this adds print time and sometimes reduces accuracy because of Z backlash.
> 
> **Describe the solution you'd like** Adding an option to this dropdown that is labelled something like 'Only when crossing ironed(both infill and post processing) surfaces'. ![image](https://user-images.githubusercontent.com/60214993/147353631-7d008084-38a1-411e-8b38-716508d87047.png)
> 
> **Describe how it would work** The slices will be able to detect if the print head is travelling over a top surface that has ironing finished and introduce zhop to that travel move
> 
> **Describe alternatives you've considered** An alternative i see is for the nozzle to take a different route rather than z hopping, however, this might not always be possible since an unfinished part might be surrounded by an ironed surface and would make re routing impossible. I see z hopping as the best solution
> 
> **Additional context** An example is this model, where the red is an ironed top layer, and the blue arrow is the scratch that will occur on that surface due to zhop being off ![image](https://user-images.githubusercontent.com/60214993/147353542-dd42cc68-2834-4d20-8377-768c4aaf9068.png)

Yes, I have experienced this and it is a very good idea if the developers do it.

**Comment by DVSVIDEO** [comment]
Yes, I have experienced this and it is a very good idea if the developers do it.

**Comment by moarkaboom** [comment]
I would also appreciate this feature. It doesn't even have to be ironed top layers, but just the top most layer in general.

**Comment by abcurrie** [comment]
I too have seen this and it would be nice to have a solution.  Ironing does not seem to be affected by selecting "avoid crossing perimeters".  I have not tried z-hopping to fix this as in general I do not use it.

**Comment by vampyrex13** [comment]
+1 on this as I prefer to leave z-hop off to prevent stringing, but I have to enable it for any of my prints that have multiple ironed surfaces.
